enhancements/20221228-various-storage-class-support.md

```diff
@@ -0,0 +1,96 @@
+# Various StorageClass support of Harvester CSI Driver
+
+## Summary
+
+Harvester supports the user-defined StorageClass after v1.1.0. Because of that, Harvester CSI Driver should also let users use the customer StorageClass mentioned above.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3052
+
+## Motivation
+
+### Goals
+
+The user using rancher integration with Harvester Cluster (Called Guest Cluster) could create the volume with the different StorageClass associated with StorageClass on the Host Harvester Cluster.
+
+## Proposal
+
+We will implement the mechanism to get the StorageClass parameter from the Guest Cluster and try to create it on the Host Harvester Cluster.
+
+### User Stories
+
```

Comment:
suggestion about user stories:

those 2 are more like motivations/goals, but not user stories

the stories are roughly aligned with test plans below, or precisely, test plans are derived from user stories

we may list a few stories like:

1. guest cluster uses default storageclass when nothing is changed

2. guest cluster gets its related storageclass manually, if UI has menu to query storageclass, add related content 

3. guest cluster may specify a customized storageclass name when cluster creating, as per detailed steps ...
  after cluster is created, any change to the storageclass has no effect (please confirm and list)
  if the storageclass name is wrong (not existing, or not in the namespace), is it checked in the validator or where?
  
 4. when a storageclass is used by a guest, is it allowed to delete this storageclass ? clarify it
 
 thanks.

Reply:
Hi @w13915984028, making user stories aligned with test plans make sense to me. I will update this again.
Thanks for the suggestions!

Reply:
Hi @w13915984028, I already updated the HEP again. Please help to review.

> 1. guest cluster uses default storageclass when nothing is changed

Add some information for it.

> 2. guest cluster gets its related storageclass manually, if UI has menu to query storageclass, add related content

Now guest cluster could not get the related StorageClass name on the host cluster because of the permission.

> 3. guest cluster may specify a customized storageclass name when cluster creating, as per detailed steps ...
>    after cluster is created, any change to the storageclass has no effect (please confirm and list)
>    if the storageclass name is wrong (not existing, or not in the namespace), is it checked in the validator or where?

We may prefer to keep the original StorageClass as the default StorageClass. That would benefit the general cases.

For the StorageClass name-checking with the host cluster. We cannot handle it well because of permission.

> 4. when a storageclass is used by a guest, is it allowed to delete this storageclass ? clarify it

Delete the related StorageClass would not affect the created PV/PVC. That I thought should be updated on StorageClass-related HEP.